src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/MissingAssemblyTests.cs

```diff
@@ -138,6 +139,56 @@ public void M(int[] array)
         }
 
         [Fact]
+        public void ERR_NoSuchMemberOrExtension()
+        {
+            var source = @"
+using System.Linq;
+
+public class C
+{
+    public void M(int[] array)
+    {
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, new[] { SystemCoreRef }, TestOptions.DebugDll);
+            var context = CreateMethodContextWithReferences(comp, "C.M", MscorlibRef);
+
+            var expectedErrorTemplate = "error CS1061: 'int[]' does not contain a definition for '{0}' and no extension method '{0}' accepting a first argument of type 'int[]' could be found (are you missing a using directive or an assembly reference?)";
+            var expectedMissingAssemblyIdentity = EvaluationContextBase.SystemCoreIdentity;
+
+            ResultProperties resultProperties;
+            string actualError;
+            ImmutableArray<AssemblyIdentity> actualMissingAssemblyIdentities;
+
+            context.CompileExpression(
+                DefaultInspectionContext.Instance,
+                "array.Count()",
+                DkmEvaluationFlags.TreatAsExpression,
+                DiagnosticFormatter.Instance,
+                out resultProperties,
+                out actualError,
+                out actualMissingAssemblyIdentities,
+                EnsureEnglishUICulture.PreferredOrNull,
+                testData: null);
+            Assert.Equal(string.Format(expectedErrorTemplate, "Count"), actualError);
+            Assert.Equal(expectedMissingAssemblyIdentity, actualMissingAssemblyIdentities.Single());
+
+            context.CompileExpression(
+                DefaultInspectionContext.Instance,
+                "array.NoSuchMethod()",
+                DkmEvaluationFlags.TreatAsExpression,
+                DiagnosticFormatter.Instance,
+                out resultProperties,
+                out actualError,
+                out actualMissingAssemblyIdentities,
+                EnsureEnglishUICulture.PreferredOrNull,
+                testData: null);
+            Assert.Equal(string.Format(expectedErrorTemplate, "NoSuchMethod"), actualError);
+            Assert.Equal(expectedMissingAssemblyIdentity, actualMissingAssemblyIdentities.Single());
```

Comment:
Correction, change the original Compilation to not include System.Core.dll. (Presumably, the RuntimeInstance does not include System.Core.dll.)